File: JavaCode/src/com/ysx/leetcode/easy/IntersectionOfTwoArrays.java
package com.ysx.leetcode.easy;

import java.util.HashSet;
import java.util.Set;

/**
 * 349. Intersection of Two Arrays
 * https://leetcode.com/problems/intersection-of-two-arrays/description/
 * 两个数组的交集
 */
public class IntersectionOfTwoArrays {

    public static void main(String[] args) {
        int[] nums1 = {1, 2, 2, 1};
        int[] nums2 = {2, 2};
        IntersectionOfTwoArrays test = new IntersectionOfTwoArrays();
        int[] nums = test.intersection(nums1, nums2);
        System.out.println("nums.length: " + nums.length);
        for (int item : nums) {
            System.out.println(item);
        }

    }

    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        for (int item : nums1){
            set1.add(item);
        }
        Set<Integer> set2 = new HashSet<>();
        for (int item : nums2) {
            if (set1.contains(item)) {
                set2.add(item);
            }
        }
        int[] result = new int[set2.size()];
        int i = 0;
        for (int item : set2){
            result[i++] = item;
        }
        return result;
    }
}

File: KotlinCode/src/leetcode/easy/IntersectionOfTwoArrays.kt
package leetcode.easy

/**
 * 349. Intersection of Two Arrays
 * https://leetcode.com/problems/intersection-of-two-arrays/description/
 * 两个数组的交集
 * Kotlin里边，Set表示不可变集合，MutableSet表示可变集合
 */
class IntersectionOfTwoArrays {
    fun intersection(nums1: IntArray, nums2: IntArray): IntArray {
        val set1: MutableSet<Int> = mutableSetOf()
        for (item in nums1) {
            set1.add(item)
        }
        val set2: MutableSet<Int> = mutableSetOf()
        for (item in nums2) {
            set2.add(item)
        }
        return set1.intersect(set2).toIntArray()
    }
}

fun main(args: Array<String>) {
    val test = IntersectionOfTwoArrays()
    val arrA = intArrayOf(1, 2, 2, 1)
    val arrB = intArrayOf(2, 2)
    val arr = test.intersection(arrA, arrB)
    for (item in arr){
        println(item)
    }
}